### PR TITLE
Split data dir, moving large files into examples/data

### LINKED ISSUE #120: [FEA] Separate morpheus data dir
**Is your feature request related to a problem? Please describe.**
Currently the Morpheus `data` dir is 270MB in size.
Some of the files in the directory (`labels_nlp.txt`, `columns_fil.txt`, `columns_ae.txt` & `bert-base-cased-hash.txt`) are quite small and are used as default values for command line flags. 

The files that are used as default values in the morpheus cli need to be copied when we perform a morpheus install (see #60 ).

**Describe the solution you'd like**
The smaller txt files should remain in the `data` dir. The larger jsonlines and csv files should be moved to somewhere under the `models` dir possibly `models/datasets`.

### PR DESCRIPTION
**Pull request by dagardner-nv**
Brings directory size down to 1.6MB down from 270MB

+ Move large files to `exaples/data` 
+ Migrate existing files in `exaples/data` to git-lfs
+ Add missing `email_with_addresses.jsonlines` needed for phishing detection developer guide
+ Update file paths in docs for data files
+ Update import paths in developer guide

Depends on changes in ~~#62~~
Fixes #120

### CODE REVIEW

**Review by raykallen — APPROVED** [review]
LGTM

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Everything looks good. I'm not sure what the `with_data_len.json` and `without_data_len.json` files are used for. Can you verify that they arent used anywhere and remove them? Otherwise, looks good.

**Comment by dagardner-nv** [comment]
Removed unused `with_data_len.json` and `without_data_len.json` files

File: docker/conda/environments/cuda11.4_dev.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ dependencies:
     - isort
     - mlflow>=1.23
     - myst-parser==0.17
-    - neo 22.04.*
+    - neo 22.04.00a*
```
Comment (mdemoret-nv) [inline docker/conda/environments/cuda11.4_dev.yml:53]: Can we unpin this? Once PR 107 goes in, this should go back to `22.04.*`

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Need to revert the pinned Neo version

**Review by mdemoret-nv — APPROVED** [review]
Looks good now.

**Pull request merged**